### ISSUE TITLE
SA1516: misfire in C# 9 top-level statements

### ISSUE THREAD

**Issue by AArnott** [DotNetAnalyzers/StyleCopAnalyzers#3351]
Given this Program.cs file:

```cs
// Copyright (c) Andrew Arnott. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using MinerForeman;

var foreman = new Foreman();
await foreman.RunAsync();
```

The last line in the file above produces:

```
Warning	SA1516	Elements should be separated by blank line	Program.cs
```

This repros with StyleCop.Analyzers 1.2.0-beta.333

**Comment by ylabade** [comment]
Any news on this issue?

**Comment by thomaslevesque** [comment]
I'm seeing this as well. Apparently _all_ top-level statements must be separated by a blank line, otherwise I have this warning.

What's weird is that the diagnostic doesn't report any file location... I have a warning, but I don't know where.

**Comment by sharwell** [comment]
@thomaslevesque The reporting location bug is #3242, and should be fixed in newer releases.

**Comment by thomaslevesque** [comment]
@sharwell thanks. Yes, I probably need to update to a newer version.

**Comment by msmolka** [comment]
What is the status of this issue? do you know maybe in which pre-release it will be done? I'm using 1.2.0-beta.406 and still seeing issue on top=level statements

**Comment by sharwell** [comment]
I don't believe anyone has submitted a fix for this issue yet.

**Comment by madhusameena** [comment]
Any comments on this ?
or any workaround for this ?

**Comment by maxkoshevoi** [comment]
> I don't believe anyone has submitted a fix for this issue yet.

@sharwell Created a fix: #3482

**Comment by maxkoshevoi** [comment]
@sharwell Thanks for merging so quickly! Is there an ETA on next NuGet release?

**Comment by domagojmedo** [comment]
Still not fixed in 1.2.0-beta.406

**Comment by ultimaweapon** [comment]
That one is released before #3482 is merged.

**Comment by sharwell** [comment]
Yep, waiting for a few more items to finish/merge and the next release should be soon.
